src/qt/buynamespage.cpp

```diff
@@ -45,22 +53,61 @@ bool BuyNamesPage::eventFilter(QObject *object, QEvent *event)
 {
     if (event->type() == QEvent::FocusIn)
     {
-        if (object == ui->registerName)
+        if (object == ui->registerNameAscii)
+        {
+            ui->registerNameButton->setDefault(true);
+        }
+
+        if (object == ui->registerNameHex)
+        {
+            ui->registerNameButton->setDefault(true);
+        }
+
+        if (object == ui->registerNameDomain)
         {
             ui->registerNameButton->setDefault(true);
         }
+
     }
     return QWidget::eventFilter(object, event);
 }
 
-void BuyNamesPage::onNameEdited(const QString &name)
-{
-    if (!walletModel)
-        return;
 
-    const QString availableError = name_available(name);
+QString BuyNamesPage::DomainToASCII(const QString &name){
+    
+    if(IsPurportedNamecoinDomain(name.toStdString()))
+    {
+        return QString::fromStdString(ASCIIFromDomain(name.toStdString()));
+    }
+    else return QString("");
+}
 
-    if (availableError == "")
+
+QString BuyNamesPage::ASCIIToDomain(const QString &name){
+    if(NamespaceFromName(name.toStdString()) == NameNamespace::Domain)
+    {
+        return QString::fromStdString(DescFromName(DecodeName(name.toStdString(), NameEncoding::ASCII), NameNamespace::Domain));
+    }
+    else return QString("");
+}
+
+
+QString BuyNamesPage::HexToASCII(const QString &name){
+    return NameTableModel::hexToAscii(name);
+}
+
+
+QString BuyNamesPage::ASCIIToHex(const QString &name){
+    return NameTableModel::asciiToHex(name);
+}
+
+void BuyNamesPage::availableError()
```

Comment:
It looks like this function has the same name as a variable inside it. The function name is also a bit confusing (it could be interpreted as something that you call when there's an error).

Can we rename this to `refreshAvailableError` or something similar?